Fix resolution of MediaType when validation errors using Accept headers

Sgitario opened the pull request:
The solution is to inject the HttpHeaders to have access to the HTTP headers in the exception mappers for Resteasy Classic and Reactive:

```java
@Context
HttpHeaders headers;
```

And then use a new utility that I called `ValidatorMediaTypeUtil` with some share functionality to get the list of media types from the `@Produces` annotation and to find the matching media type to serialize the validation reports.

Fix https://github.com/quarkusio/quarkus/issues/20859

yrodiere reviewed (changes requested):
Thanks! The solution looks good to me overall. I added a few suggestions, though.

Sgitario committed:
Add reflective class ConstraintType.Type in Resteasy JAXB extension

geoand reviewed (changes requested):
I suggest limiting the change to RESTEasy Classic for now. 
We don't want to have to do reflection in the case of RESTEasy Reactive.
Let's get this on for now and talk about the proper RESTEasy Reactive solution after

Sgitario committed:
Fix resolution of MediaType when validation errors using Accept headers

The solution is to inject the HttpHeaders to have access to the HTTP headers in the exception mappers for only Resteasy Classic:

```java
@Context
HttpHeaders headers;
```

And then use a new utility that I called `ValidatorMediaTypeUtil` to find the matching media type to serialize the validation reports.
